src/arena/wall.py

```diff
@@ -29,5 +31,12 @@ def init_for_drawing(self):
             int(self._size[0]), int(self._size[1]), color.BLACK
         )
         self._sprite.owner = self
+        self.upper_left: tuple = (self._sprite.left, self._sprite.top)
+        self.upper_right: tuple = (self._sprite.right, self._sprite.top)
+        self.bottom_left: tuple = (self._sprite.left, self._sprite.bottom)
+        self.bottom_right: tuple = (self._sprite.right, self._sprite.bottom)
+        self.list_of_corners: list[tuple](
+            self.upper_left, self.upper_right, self.bottom_right, self.bottom_left
+        )
```

Comment:
I pinpointed what's throwing the error when the car drives into the wall. missing equals sign.
self.list_of_corners: list[tuple] = (
            self.upper_left, self.upper_right, self.bottom_right, self.bottom_left
        )
Once fixed, the car still drives through the diagonal wall, but doesn't throw an error